app/frontend/src/lib/ui/components/panel.test.js

```diff
@@ -0,0 +1,77 @@
+import panel, { toggleButtonText, isPanelClosed, togglePanel } from './panel';
+
+describe('panel', () => {
+  document.body.innerHTML = '<div id="panel-container"></div><button id="toggle-button">original text</button>';
+
+  const HIDE_BUTTON_TEXT = 'Hide panel';
+  const SHOW_BUTTON_TEXT = 'Show panel';
+  const CLOSED_CLASS = 'closed';
+  const container = document.getElementById('panel-container');
+  const button = document.getElementById('toggle-button');
+  let options = null;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    options = {
+      container,
+      toggleButton: button,
+      hideText: HIDE_BUTTON_TEXT,
+      showText: SHOW_BUTTON_TEXT,
+      toggleClass: CLOSED_CLASS,
+      onToggleHandler: jest.fn(),
+      onClosedHandler: jest.fn(),
+      onOpenedHandler: jest.fn(),
+    };
+  });
+
+  describe('toggleButtonText', () => {
+    test('changes button text to show panel state when panel is hidden', () => {
```

Comment:
I'm not sure this message matches up with line 31 properly? Or maybe I'm just being confused today 😂